src/test/java/de/tum/cit/aet/artemis/communication/ReactionIntegrationTest.java

```diff
@@ -118,18 +117,20 @@ void tearDown() {
     void testCreateOwnPostReaction(CourseInformationSharingConfiguration courseInformationSharingConfiguration, boolean shouldBeAllowed) throws Exception {
         // student 1 is the author of the post and reacts on this post
         Post postReactedOn = existingPostsWithAnswers.getFirst();
-        Reaction reactionToSaveOnPost = createReactionOnPost(postReactedOn);
+        ReactionDTO reactionToSaveOnPost = createReactionDTOOnPost(postReactedOn);
 
         course.setCourseInformationSharingConfiguration(courseInformationSharingConfiguration);
         courseRepository.save(course);
 
         if (!shouldBeAllowed) {
-            request.postWithResponseBody("/api/communication/courses/" + courseId + "/postings/reactions", reactionToSaveOnPost, Reaction.class, HttpStatus.BAD_REQUEST);
+            request.postWithResponseBody("/api/communication/courses/" + courseId + "/postings/reactions", reactionToSaveOnPost, ReactionDTO.class, HttpStatus.BAD_REQUEST);
             return;
         }
 
-        Reaction createdReaction = request.postWithResponseBody("/api/communication/courses/" + courseId + "/postings/reactions", reactionToSaveOnPost, Reaction.class,
+        // Act: Send valid reaction
+        ReactionDTO createdReaction = request.postWithResponseBody("/api/communication/courses/" + courseId + "/postings/reactions", reactionToSaveOnPost, ReactionDTO.class,
```

Comment:
No strong opinion, keep it in if you think it adds value - from my perspective it doesn't
```suggestion
        ReactionDTO createdReaction = request.postWithResponseBody("/api/communication/courses/" + courseId + "/postings/reactions", reactionToSaveOnPost, ReactionDTO.class,
```